### PR TITLE
CMake modern target usage

### PR DESCRIPTION
**Pull request by NeroBurner**
Instead of variables use target_sources to add source files to an existing target.
Support multiple versions of SDL2 imported targets

Edit: Hope the spaced instead of tabs don't shock or anger you :see_no_evil:

### CODE REVIEW

**Comment by flibitijibibo** [comment]
The tabs/spaces thing is fixable :P

The changes are fine but if they can be used by something first (utils folder?) then we'll know that the changes are meaningful too. See if this makes it any easier to build/run the utils; one thing I totally forgot about was the wavcommon paths:

https://github.com/FNA-XNA/FAudio/blob/master/utils/wavcommon/wavs.cpp#L4

**Comment by NeroBurner** [comment]
updated, I can squash the commits if you'd like

I could have used globbing for the util executable, but I like explicit naming of files for cmake targets (also the header files, because then they show up in the IDE)

File: CMakeLists.txt [hunk 1]
```diff
@@ -19,44 +19,60 @@ endif()
...
+elseif (TARGET SDL2)
+    message(STATUS "using TARGET SDL2")
+    target_link_libraries(FAudio PUBLIC SDL2)
+else()
+    message(STATUS "no TARGET SDL2::SDL2, or SDL2, using variables")
+    target_include_directories(FAudio PUBLIC ${SDL2_INCLUDE_DIRS})
+    target_link_libraries(FAudio PUBLIC ${SDL2_LIBRARIES})
+endif()
 
 # TODO: utils/
```
Comment (flibitijibibo) [inline CMakeLists.txt]: This should take care of the utils TODO. Are we able to folder the other CMakeLists.txt into this one? I try not to have nested CMake files unless the projects are actually separate from one another in a significant way.

Other than that, spaces->tabs should wrap this up!

**Comment by NeroBurner** [comment]
for the wavs RESOURCE_PATH i'd recommend setting it by macro from cmake and only define it hard coded if it is not yet defined
```cpp
#ifndef RESOURCE_PATH
#ifdef _MSC_VER
#define RESOURCE_PATH "../../utils/wavcommon/resources"
#else
#define RESOURCE_PATH "utils/wavcommon/resources"
#endif
#endif 
```

**Comment by flibitijibibo** [comment]
That'd work for me!

File: CMakeLists.txt (hunk 1 above)
Comment (NeroBurner) [reply CMakeLists.txt]: sure, I don't like to have huge CMakeLists files, but it is your repo, so your wish is my command :)

Comment (NeroBurner) [reply CMakeLists.txt]: I remember why I like subfolder cmake lists files. I now have to add `utils/` before every source file

**Comment by NeroBurner** [comment]
squashed and modified `wavs.cpp`

**Comment by flibitijibibo** [comment]
Excellent, thanks for looking at this!

**Comment by NeroBurner** [comment]
you're welcome :+1: 
Thanks for working on FAudio. Thanks to you (and all working on FAudio) I can play Warframe in 64bit mode ^_^

**Pull request merged**